README.md

```diff
@@ -1,19 +1,136 @@
 # xKAN-meta_testing
-Contains the build scripts used by Jenkins
-
-Contributions
-=============
-
-These were moved over from KSP-CKAN/CKAN-meta and KSP-CKAN/NetKAN. Thanks
-to all the contributions from the following in no particular order.
-
-Matthew Heguy  
-Leon Wright  
-Dwayne Bent  
-Willhelm Rendahl  
-Paul Fenwick  
-Alexander Dzhoganov  
-Magnus Aagaard Sørensen  
-Hakan Tandogan  
-Myk Dowling  
-Arne Peirs  
+
+GitHub Action to validate .netkan and .ckan files in GitHub repos.
+
+## Usage
+
+If you are the author or maintainer of a mod that uses a [meta-netkan], you can use this Action to validate your mod's CKAN metadata. It will appear in the Actions tab of your repo on GitHub and automatically parse your .netkan file, generate a .ckan file based on the current release, and install your mod and its dependencies in a sandbox, reporting any problems it finds along the way. Note that .ckan files will be ignored in your repo because they might be [internal .ckan files] lacking enough info to inflate or validate.
+
+[meta-netkan]: https://github.com/KSP-CKAN/CKAN/blob/master/Spec.md#ckannetkanurl
+[internal .ckan files]: https://github.com/KSP-CKAN/CKAN/wiki/Adding-a-mod-to-the-CKAN#internal-ckan-files
+
+Create a file in your repo at `.github/workflows/netkan.yml` as follows:
+
+```yml
+on:
+    push:
+        branches:
+            - master
+    pull_request:
+        types:
+            - opened
+            - synchronize
+            - reopened
+jobs:
+    Inflate:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Get mod repo
+              uses: actions/checkout@v2
+            - name: Test meta-netkans
+              uses: KSP-CKAN/xKAN-meta_testing@master
+              with:
+                  pull request body: ${{ github.event.pull_request.body }}
+```
+
+If you're on the CKAN dev team, a few changes are needed for the main metadata repos, to make the Action search the commit history for changes:
+
+- The checkout action needs `fetch-depth: 0` to get the full commit history
```

Comment:
What are we using the full commit history for? There's a fair amount of work in pulling the full tree.

Reply:
Initially I was not able to determine which files were changed on the pull request without that. But that was with an earlier iteration of that code (putting all the SHAs from `master` in a hash and using it to filter rather than @DasSkelett's `.iter_commits` suggestion), so it may or may not be needed anymore.